### DISCUSSION TITLE
Version compatibility for devops-infra/docker-terragrunt

### DISCUSSION THREAD

**Discussion by mehdicopter** [renovatebot/renovate#27771] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

Hello I am using the image from [devops-infra/docker-terragrunt](https://github.com/devops-infra/docker-terragrunt) repository within my gitlab-ci.yml file like so.

```yaml
# .gitlab-ci.yml
default:
  image:
    name: devopsinfra/docker-terragrunt:gcp-tf-1.7.3-tg-0.55.2
```

I want to be able to catch when `1.7.3` and `0.55.2` can be bumped.
Checking the documentation I need to modify `versionCompatibility`
Is it possible to catch multiple SemVer in the same string?
I have this configuration but it does not work.

```json
"gitlabci": {
    "packageRules": [
      {
        "matchPackageNames": "devopsinfra/docker-terragrunt",
        "versionCompatibility": "regex:^(?<compatibility>.*)-(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)-.*?$"
      }
    ]
  }
```

What am I doing wrong ? :(

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: version compatibility regex mismatch (repository=xxxx/test-renovate)
       "versionCompatibility": "regex:^(?<compatibility>.*)-(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)-.*?$",
       "currentValue": "gcp-tf-1.7.3-tg-0.55.2",
       "packageName": "devopsinfra/docker-terragrunt"
DEBUG: Dependency devopsinfra/docker-terragrunt has unsupported/unversioned value gcp-tf-1.7.3-tg-0.55.2 (versioning=docker) (repository=xxxx/test-renovate)
```

</details>

**Comment by rarkins** [comment]
Can you give some examples of what are valid and invalid upgrades? Or is every higher version ok no matter what?

**Reply by mehdicopter** [reply]
Sure,
for instance the latest image available has this tag: `gcp-tf-1.7.4-tg-0.55.12`
I did try to set
```json
"gitlabci": {
    "packageRules": [
      {
        "matchPackageNames": "devopsinfra/docker-terragrunt",
        "versioning": "loose"
      }
    ]
  }
```
but it did not work as well

**Reply by rarkins** [reply]
Use versionCompatibility to select the prefix as compatibility and everything from the first digit as the version. Then rely on loose versioning to sort them

**Reply by mehdicopter** [reply]
Do you mean I should mix both configurations I gave above ?

**Reply by mehdicopter** [reply]
```
DEBUG: version compatibility regex mismatch (repository=xxxx/test-renovate)
       "versionCompatibility": "regex:^(?<compatibility>.*)-(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)-.*?$",
       "currentValue": "gcp-tf-1.7.3-tg-0.55.2",
       "packageName": "devopsinfra/docker-terragrunt"
DEBUG: Dependency devopsinfra/docker-terragrunt has unsupported/unversioned value gcp-tf-1.7.3-tg-0.55.2 (versioning=loose) (repository=xxxx/test-renovate)

**Reply by rarkins** [reply]
That's not how you use versionCompatibility

**Reply by mehdicopter** [reply]
Could you please elaborate or even better maybe giving me the solution to my problem ? 😇

**Reply by rarkins** [reply]
The docs section for versionCompatibility contains examples

**Reply by mehdicopter** [reply]
thanks for this RTFM answer :)

**Comment by BlueIce** [comment, marked as answer]
This package rule worked for me (based on https://github.com/renovatebot/renovate/discussions/27771#discussioncomment-8706250):

```json
{
    "matchPackageNames": ["devopsinfra/docker-terragrunt"],
    "versionCompatibility": "^(?<compatibility>[^\\d]*)(?<version>.*)$",
    "versioning": "loose"
}
```
